src/test/java/uk/gov/di/ipv/cri/common/library/domain/personidentity/AddressTest.java

```diff
@@ -9,24 +9,92 @@
 
 class AddressTest {
     @Test
-    void shouldGetPastAddressType() {
+    void shouldGetPreviousAddressTypeWhenValidUntilIsPastDate() {
         Address testAddress = new Address();
         testAddress.setValidUntil(LocalDate.of(2013, 8, 9));
 
         assertEquals(AddressType.PREVIOUS, testAddress.getAddressType());
     }
 
     @Test
-    void shouldGetCurrentAddressType() {
+    void shouldGetPreviousAddressTypeWithValidUntilIsToday() {
+        Address testAddress = new Address();
+        testAddress.setValidUntil(LocalDate.now());
+
+        assertEquals(AddressType.PREVIOUS, testAddress.getAddressType());
+    }
+
+    @Test
+    void shouldReturnNullAddressTypeWhenValidUntilIsInFuture() {
+        Address testAddress = new Address();
+        testAddress.setValidUntil(LocalDate.now().plusYears(1));
+
+        assertNull(testAddress.getAddressType());
+    }
+
+    @Test
+    void shouldGetPreviousAddressTypeWithValidUntilAndValidFrom() {
+        Address testAddress = new Address();
+        testAddress.setValidFrom(LocalDate.of(2013, 7, 9));
+        testAddress.setValidUntil(LocalDate.of(2013, 8, 9));
+
+        assertEquals(AddressType.PREVIOUS, testAddress.getAddressType());
+    }
+
+    @Test
+    void shouldGetCurrentAddressTypeWithValidFromInPast() {
         Address testAddress = new Address();
         testAddress.setValidFrom(LocalDate.of(2021, 9, 10));
 
         assertEquals(AddressType.CURRENT, testAddress.getAddressType());
     }
 
     @Test
-    void shouldReturnNullAddressTypeWhenValidFromAndValidUntilNotPresent() {
+    void shouldReturnCurrentAddressTypeWhenValidFromAndValidUntilAreNull() {
+        Address testAddress = new Address();
+
+        assertEquals(AddressType.CURRENT, testAddress.getAddressType());
+    }
+
+    @Test
+    void shouldReturnCurrentAddressTypeWhenValidFromIsToday() {
+        Address testAddress = new Address();
+        testAddress.setValidFrom(LocalDate.now());
+
+        assertEquals(AddressType.CURRENT, testAddress.getAddressType());
+    }
+
+    @Test
+    void shouldReturnNullAddressTypeWhenValidFromIsInFuture() {
+        Address testAddress = new Address();
+        testAddress.setValidFrom(LocalDate.now().atTime(0, 0).toLocalDate().plusYears(1));
```

Comment:
this `atTime` invocation can also be removed and you can simply refer to `LocalDate.now()`

Reply:
Somehow missed that one when removing the others.